tests/MongoDB.Driver.Tests/Specifications/auth/AuthTestRunner.cs

```diff
@@ -0,0 +1,147 @@
+/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core.Authentication;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.auth
+{
+    public class AuthTestRunner
+    {
+        [SkippableTheory]
+        [ClassData(typeof(TestCaseFactory))]
+        public void RunTestDefinition(JsonDrivenTestCase testCase)
+        {
+            var definition = testCase.Test;
+            JsonDrivenHelper.EnsureAllFieldsAreValid(definition, "description", "uri", "valid", "credential");
+
+            MongoCredential mongoCredential = null;
+            Exception parseException = null;
+            try
+            {
+                var connectionString = (string)definition["uri"];
+                mongoCredential = MongoClientSettings.FromConnectionString(connectionString).Credential;
+            }
+            catch (Exception ex)
+            {
+                parseException = ex;
+            }
+
+            if (parseException == null)
+            {
+                AssertValid(mongoCredential, definition);
+            }
+            else
+            {
+                AssertInvalid(parseException, definition);
+            }
+        }
+
+        private void AssertValid(MongoCredential mongoCredential, BsonDocument definition)
+        {
+            if (!definition["valid"].ToBoolean())
+            {
+                throw new AssertionException($"The connection string '{definition["uri"]}' should be invalid.");
+            }
+
+            var credential = definition["credential"] as BsonDocument;
```

Comment:
Consider renaming `credential` to `expectedCredential`.

Reply:
Done